src/main.rs

```diff
@@ -23,18 +23,18 @@ struct Sparkle {
 }
 
 impl Sparkle {
-    fn draw(&self) {
-        // each sparkle is pure white
-        let (r, g, b) = (1.0, 1.0, 1.0);
-
-        // simple sine-wave for sparkling
+    fn draw(&self, color: Color) {
+        // Simple sine-wave for sparkling
         let alpha = (self.cycle.sin() + 1.0) / 2.0;
 
-        // draw the circle
-        let color = Color::new(r, g, b, alpha);
+        // Set the color with varying alpha
+        let mut color = color;
+        color.a = alpha;
+
+        // Draw the circle
         draw_circle(self.position.x, self.position.y, self.radius, color);
     }
-}
+}    
```

Comment:
extra whitespace here

Reply:
Fixed the nits as suggested. Thanks for the review!